src/jaxsim/api/model.py

```diff
@@ -1931,11 +1932,17 @@ def step(
         ),
     )
 
+    exceptions.raise_if(
+        condition=t0_ns + jnp.array(dt * 1e9).astype(t0_ns.dtype) < t0_ns,
+        exception=OverflowError,
+        msg="The simulation time overflowed the maximum integer value. Consider using x64 by setting `JAX_ENABLE_X64=1`.",
+    )
+
     data_tf = (
         # Store the new state of the model and the new time.
         data.replace(
             state=state_tf,
-            time_ns=t0_ns + jnp.array(dt * 1e9).astype(jnp.uint64),
+            time_ns=t0_ns + jnp.array(dt * 1e9).astype(int),
```

Comment:
```suggestion
            time_ns=t0_ns + jnp.array(dt * 1e9, dtype=t0_ns.dtype),
```

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/225/commits/051704a3690fb977a1f6e666b7846da90be5fb02